### PR TITLE
feat: application commands

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: pkg/management/api/grpc/application_converter.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	for i, responseType := range responseTypes {
+		switch responseType {
+		case proj_model.OIDCRESPONSETYPE_CODE:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_CODE
+		case proj_model.OIDCRESPONSETYPE_ID_TOKEN:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_ID_TOKEN
+		case proj_model.OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			oidcResponseTypes[i] = OIDCResponseType_OIDCRESPONSETYPE_TOKEN_ID_TOKEN
+		default:
+			oidcResponseTypes[i] = OIDCResponseType(99)
```
Comment (adlerhurst) [inline pkg/management/api/grpc/application_converter.go:133]: isn't that `UNSPECIFIED`?

File: pkg/management/api/grpc/application_converter.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+	for i, grantType := range grantTypes {
+		switch grantType {
+		case OIDCGrantType_OIDCGRANTTYPE_AUTHORIZATION_CODE:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_AUTHORIZATION_CODE
+		case OIDCGrantType_OIDCGRANTTYPE_IMPLICIT:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_IMPLICIT
+		case OIDCGrantType_OIDCGRANTTYPE_REFRESH_TOKEN:
+			oidcGrantTypes[i] = proj_model.OIDCGRANTTYPE_REFRESH_TOKEN
+		}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/application_converter.go:151]: add default

File: pkg/management/api/grpc/application_converter.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+	for i, responseType := range responseTypes {
+		switch responseType {
+		case OIDCResponseType_OIDCRESPONSETYPE_CODE:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_CODE
+		case OIDCResponseType_OIDCRESPONSETYPE_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_ID_TOKEN
+		case OIDCResponseType_OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			oidcResponseTypes[i] = proj_model.OIDCRESPONSETYPE_TOKEN_ID_TOKEN
+		}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/application_converter.go:115]: add default

File: cmd/zitadel/startup.yaml [hunk 4]
```diff
@@ -27,6 +27,11 @@ Mgmt:
           User: 'management'
           Database: 'management'
           SSLmode: disable
+      Cache:
+        Type: 'fastcache'
+        ID: 'fastcache'
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply cmd/zitadel/startup.yaml]: this was befor i merged master into this branch

File: pkg/management/api/grpc/application_converter.go (hunk 1 above)
Comment (fgerschwiler) [reply pkg/management/api/grpc/application_converter.go:133]: this cant actualy happen, i removed it

File: pkg/management/api/grpc/application_converter.go (hunk 2 above)
Comment (fgerschwiler) [reply pkg/management/api/grpc/application_converter.go:151]: i think that doesnt make sense, it should be one of these 3 types if something else happens i dont want to do something

File: pkg/management/api/grpc/application_converter.go (hunk 3 above)
Comment (fgerschwiler) [reply pkg/management/api/grpc/application_converter.go:115]: i think that doesnt make sense, it should be one of these 3 types if something else happens i dont want to do something

File: internal/project/model/application.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		case OIDCRESPONSETYPE_CODE:
+			code = true
+		case OIDCRESPONSETYPE_ID_TOKEN:
+			idToken = true
+		case OIDCRESPONSETYPE_TOKEN_ID_TOKEN:
+			tokenIdToken = true
+		}
+	}
+	return code, idToken, tokenIdToken
+}
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/project/model/application.go:134]: do you have a better idea that it doesnt looks so ugly, if one is 2 in the list the func will have the same result

File: pkg/management/api/grpc/application_converter.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+func oidcApplicationTypeToModel(appType OIDCApplicationType) proj_model.OIDCApplicationType {
+	switch appType {
+	case OIDCApplicationType_OIDCAPPLICATIONTYPE_WEB:
+		return proj_model.OIDCAPPLICATIONTYPE_WEB
+	case OIDCApplicationType_OIDCAPPLICATIONTYPE_USER_AGENT:
+		return proj_model.OIDCAPPLICATIONTYPE_USER_AGENT
+	case OIDCApplicationType_OIDCAPPLICATIONTYPE_NATIVE:
+		return proj_model.OIDCAPPLICATIONTYPE_NATIVE
+	}
+	return 99
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply pkg/management/api/grpc/application_converter.go]: done

File: pkg/management/api/grpc/application_converter.go [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+func oidcApplicationTypeFromModel(appType proj_model.OIDCApplicationType) OIDCApplicationType {
+	switch appType {
+	case proj_model.OIDCAPPLICATIONTYPE_WEB:
+		return OIDCApplicationType_OIDCAPPLICATIONTYPE_WEB
+	case proj_model.OIDCAPPLICATIONTYPE_USER_AGENT:
+		return OIDCApplicationType_OIDCAPPLICATIONTYPE_USER_AGENT
+	case proj_model.OIDCAPPLICATIONTYPE_NATIVE:
+		return OIDCApplicationType_OIDCAPPLICATIONTYPE_NATIVE
+	}
+	return 99
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply pkg/management/api/grpc/application_converter.go]: done

File: internal/project/repository/eventsourcing/oidc.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	if err != nil {
+		logging.Log("APP-UpnTI").OnError(err).Error("unable to create client secret")
+		return "", nil, errors.ThrowInternal(err, "APP-gH2Wl", "unable to create password")
+	}
+	secret, err := crypto.Hash([]byte(stringSecret), alg)
+	if err != nil {
+		return "", nil, errors.ThrowInternal(err, "APP-gH2Wl", "unable to hash password")
+	}
+	return stringSecret, secret, nil
+}
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/project/repository/eventsourcing/oidc.go]: done

File: internal/project/model/application.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/project/model/application.go]: as i wrote the same way like `containsGrantType`

File: internal/project/repository/eventsourcing/model.go [hunk 9]
```diff
@@ -196,12 +450,165 @@ func (p *Project) appendRemoveMemberEvent(event *es_models.Event) error {
...
+	}
+	for i, a := range p.Applications {
+		if a.AppID == config.AppID {
+			p.Applications[i].OIDCConfig.getData(event)
+		}
+	}
+	return nil
+}
+
+func (o *OIDCConfig) getData(event *es_models.Event) error {
```
Comment (adlerhurst) [inline internal/project/repository/eventsourcing/model.go:607]: shouldn't it be appendEvents?

File: internal/project/model/application.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/project/model/application.go]: done

File: internal/project/model/oidc_config.go [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+		case OIDCRESPONSETYPE_CODE:
+			code = true
+		case OIDCRESPONSETYPE_ID_TOKEN:
+			idToken = true
+		case OIDCRESPONSETYPE_TOKEN:
+			tokenIdToken = true
+		}
+	}
+	return code, idToken, tokenIdToken
+}
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/project/model/oidc_config.go:92]: this would not make what i need

File: internal/errors/caos_error.go [hunk 11]
```diff
@@ -26,9 +26,9 @@ func CreateCaosError(parent error, id, message string) *CaosError {
 
 func (err *CaosError) Error() string {
 	if err.Parent != nil {
-		return fmt.Sprintf("ID=%s Message=%s Parent=(%v)", err.ID, err.Message, err.Parent)
+		return fmt.Sprintf("AggregateID=%s Message=%s Parent=(%v)", err.ID, err.Message, err.Parent)
```
Comment (adlerhurst) [inline internal/errors/caos_error.go:29]: you renamed too much :D

File: internal/errors/caos_error.go [hunk 12]
```diff
@@ -26,9 +26,9 @@ func CreateCaosError(parent error, id, message string) *CaosError {
...
 	}
-	return fmt.Sprintf("ID=%s Message=%s", err.ID, err.Message)
+	return fmt.Sprintf("AggregateID=%s Message=%s", err.ID, err.Message)
```
Comment (adlerhurst) [inline internal/errors/caos_error.go:31]: you renamed too much :D

File: internal/errors/caos_error_test.go [hunk 13]
```diff
@@ -10,10 +10,10 @@ import (
 
 func TestErrorMethod(t *testing.T) {
 	err := caos_errs.ThrowError(nil, "id", "msg")
-	expected := "ID=id Message=msg"
+	expected := "AggregateID=id Message=msg"
```
Comment (adlerhurst) [inline internal/errors/caos_error_test.go:13]: you renamed too much :D

File: internal/errors/caos_error_test.go [hunk 14]
```diff
@@ -10,10 +10,10 @@ import (
...
 	assert.Equal(t, expected, err.Error())
 
 	err = caos_errs.ThrowError(err, "subID", "subMsg")
-	subExptected := "ID=subID Message=subMsg Parent=(ID=id Message=msg)"
+	subExptected := "AggregateID=subID Message=subMsg Parent=(AggregateID=id Message=msg)"
```
Comment (adlerhurst) [inline internal/errors/caos_error_test.go:17]: you renamed too much :D

**Pull request merged**